lstchain/calib/camera/pixel_threshold_estimation.py

```diff
@@ -79,3 +83,77 @@ def get_unusable_pixels(dl1_path, pedestal_id):
+    hence pixels illuminated by stars get a higher threshold, and so we avoid
+    too many spurious signals from the starlight fluctuations. The downside of
+    the method is that it introduces non-uniformities in the camera response
+    for the real data, and therefore data-MC discrepancies (there are no stars
+    in MC and all pixels have the same cleaning settings).
+
+    Here we try to calculate what should be the "base" picture threshold (to
+    use both in MC and data) so that at most a given fraction "max_fraction"
+    of the camera gets an increased threshold via the
+    clean_with_pedestal_threshold condition. In this way the number /
+    extension of camera inhomogeneities in the real data is limited. By
+    default the max_fraction is 0.04, which e.g. gives a picture threshold
+    around 8 for the Crab field (in no-moon conditions).
+
+    Note: we want cleaning settings for a whole run, so we will have to run
+    the function over sub-runs and average the values, or make the function
+    able to read multiple DL1 files
+
+    Parameters
+    ----------
+    dl1_path: real data DL1 file, to have access to the monitoring table
+    where we can read the pedesta bias & std dev estimated with interleaved
+    pedestal events
+
+    config_file: must be the one used for the analysis of the real data,
+    in particular, it has to contain the tailcuts_clean_with_pedestal_threshold
+    setting (sigma) that one wants to use
+
+    max_fraction: maximum fraction of camera pixels that are allowed to get a
+    picture threshold above the base one. That is, we calculate here the base
+    picture threshold that will ensure that the condition is fulfilled
+
+    Returns
+    -------
+    (scalar) the value of the picture threshold that has to be used in data and
+    MC to ensure that no more than max_fraction of the camera gets an
+    increased value via tailcuts_clean_with_pedestal_threshold
+
+    """
+    std_config = get_standard_config()
+    if config_file is not None:
+        config = replace_config(std_config,
+                                read_configuration_file(config_file))
+    else:
+        config = std_config
+
+    cleaning_method = 'tailcuts_clean_with_pedestal_threshold'
+    picture_th, _, _, _ = get_cleaning_parameters(config, cleaning_method)
+    sigma = config[cleaning_method]['sigma']
+
+    # Obtain the picture thresholds of pixels based on the "clean with
+    # pedestal threshold" method:
+    pic_threshold = get_threshold_from_dl1_file(dl1_path, sigma)
+    threshold_sorted = np.sort(pic_threshold)
+
+    # find the value new_threshold above which a fraction max_fraction of
+    # pixels lies:
+    index = int(len(threshold_sorted) * (1 - max_fraction))
+    new_threshold = threshold_sorted[index]
+
+    # Return the first integer value above new_threshold (to avoid too many
+    # different cleaning settings in different runs):
+
```

Comment:
why is it a problem to get different cleaning settings in different runs? with this `ceil` we can also have very different settings if we have `new_threshold` = 8.9 for one run and 9.1 for the next one, even in the case they are not that different

Reply:
> why is it a problem to get different cleaning settings in different runs? with this `ceil` we can also have very different settings if we have `new_threshold` = 8.9 for one run and 9.1 for the next one, even in the case they are not that different

Yes, of course, the rounding does not avoid steps. The comment refers to rounding as opposed to using a floating-point value.

It is a problem for everything which is not just "produce run-wise IRFs". For that, indeed, we don't care if each run has different settings because it will have different IRF, and the corresponding MC can be produced with the same cleanings. But for other things, like e.g. data-MC comparisons, we have to put together many runs, and compare to a given MC set (the same MC for different reals runs, like in case of data taken on consecutive nights on the Crab in the same zenith range), and definitely we want to have an homogeneous analysis. The use of integer cleanings means that at least of conditions are the same (NSB etc) chances are that the bulk of the data for a given source and observation direction are processed in a homogeneous way.